Java/core/src/main/java/com/amazon/randomcutforest/tree/Node.java

```diff
@@ -75,7 +76,7 @@
      * RandomCutTree, this set stores the indexes corresponding to times when the
      * given leaf point was added to the tree.
      */
-    private Set<Long> sequenceIndexes;
+    private HashMap<Long, Integer> sequenceIndexes;
```

Comment:
minor: We can make the field type `Map` instead of the concrete type `HashMap`.

Reply:
Done.